src/Testcontainers.Xunit/DbContainerTestMethods.cs

```diff
@@ -0,0 +1,41 @@
+namespace DotNet.Testcontainers.Xunit;
+
+internal class DbContainerTestMethods(DbProviderFactory dbProviderFactory, string connectionString) : IDbContainerTestMethods, IAsyncDisposable
```

Comment:
The interface `IDbContainerTestMethods` should inherit from `IDisposable` and `IAsyncDisposable` instead of the class `DbContainerTestMethods`. This ensures that clean disposal takes place in every scenario.

Reply:
The `IDbContainerTestMethods` (internal) interface only purpose is to share the documentation of its methods between `DbContainerFixture` and `DbContainerTest`, so that  `/// <inheritdoc />` can be used instead of duplicating the documentation.

Anyway, it would be more complicated than necessary since both `DbContainerFixture` and `DbContainerTest` implement xUnit's `IAsyncLifetime` interface whose `DisposeAsync` method returns a `Task` instead of a `ValueTask` (which is the return type used by `IAsyncDisposable`). And both `DbContainerFixture` and `DbContainerTest` already properly call `_testMethods.DisposeAsync()`.

But your comment raised a good point. The `IAsyncLifetime` interface should be implemented explicitly. I'll fix it, rebase onto the `develop` branch and force push.